### PR TITLE
Ready: Add the Yokozuna Index Creation timeout parameter for Riak 2.1+

### LINKED ISSUE #613: Support synchronous YZ index creation. [JIRA: CLIENTS-826]
Right now, if you create a Yokozuna index it's basically an asynchronous call. Riak responds immediately and at some point in the future the index will be created.

Because this is somewhat problematic, @zeeshanlakhani changed it so that it will block until the index is known to have been created, or a timeout occurs. This can either be the default timeout (60 seconds) or a timeout specified by the user sent with the create index request. The timeout doesn't mean the index hasn't been created, but rather the confirmation wasn't received.

This was included in Riak 2.1.

CLIENTS-333

### PR DESCRIPTION
**Pull request by alexmoore**
Fixes #613 (CLIENTS-826).

### CODE REVIEW

File: src/test/java/com/basho/riak/client/core/operations/itest/ISearchTestBase.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ private static void setupYokoIndexAndBucketProps(RiakClient client, Namespace na
             throws ExecutionException, InterruptedException
     {
         YokozunaIndex index = new YokozunaIndex(indexName);
-        StoreIndex ssi = new StoreIndex.Builder(index).withTimeout(45).build();
+        StoreIndex ssi = new StoreIndex.Builder(index).withTimeout(45000).build();
```
Comment (srgg) [inline src/test/java/com/basho/riak/client/core/operations/itest/ISearchTestBase.java]: Wy don't us make this timeout to be customizable, and use this value as a default, not as a  Magic Number but as a Symbolic Constant like DEFAULT_STORE_INDEX_TIMEOUT

Comment (alexmoore) [inline src/test/java/com/basho/riak/client/core/operations/itest/ISearchTestBase.java]: I got tired last night and didn't make it that far :)

**Comment by srgg** [comment]
👍 LGTM

**Comment by christophermancini** [comment]
👍 Tests pass, code looks good.

**Pull request merged**